extra_geom/detectors.py

```diff
@@ -930,67 +1000,102 @@ def inspect(self, axis_units='px', frontview=True):
         ax.set_title('LPD-1M detector geometry ({})'.format(self.filename))
         return ax
 
-    @staticmethod
-    def split_tiles(module_data):
-        # This slicing is faster than using np.split()
-        return [
-            # Tiles 1-8 numbered top to bottom. Data starts at bottom, so
-            # count backwards through them.
-            module_data[..., y-32:y, :128] for y in range(256, 0, -32)
-        ] + [
-            # Tiles 9-16 numbered bottom to top.
-            module_data[..., y:y+32, 128:] for y in range(0, 256, 32)
-        ]
 
-    @classmethod
-    def _tile_slice(cls, tileno):
-        # Which part of the array is this tile?
-        if tileno < 8:  # First half of module (0 <= t <= 7)
-            fs_slice = slice(0, 128)
-            tiles_up = 7 - tileno
-        else:  # Second half of module (8 <= t <= 15)
-            fs_slice = slice(128, 256)
-            tiles_up = tileno - 8
-        tile_offset = tiles_up * 32
-        ss_slice = slice(tile_offset, tile_offset + cls.frag_ss_pixels)
-        return ss_slice, fs_slice
+class LPD_MiniGeometry(LPDGeometryBase):
+    """Detector Layout for LPD Mini.
+
+    The coordinates used in this class are 3D (x, y, z), and represent metres.
+
+    You won't normally instantiate this class directly:
+    use one of the constructor class methods to create or load a geometry.
+    """
+    detector_type_name = 'LPD-Mini'
+    n_quads = 1
+    n_modules = 1
+    pixel_size = 5e-4  # 5e-4 metres == 0.5 mm
+    expected_data_shape = (1, 256, 256)
```

Comment:
I guess we might want to allow the LPD Mini class to handle 2D input arrays. We already do something similar for ePix - have a look at the `_ensure_shape` method.